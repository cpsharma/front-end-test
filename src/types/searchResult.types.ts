export interface OptionsType {
	label: string;
	value: string;
};

export interface RatingAndFacilitiesType extends OptionsType {
	isChecked: boolean;
	name: string;
};

export interface PricePerPersonType extends RatingAndFacilitiesType {
	min: string;
	max: string;
}