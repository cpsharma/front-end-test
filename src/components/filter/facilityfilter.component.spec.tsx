import { h, JSX } from "preact";
import { shallow, mount, configure, ShallowWrapper, cleanup } from "enzyme";
import FacilityFilterComponent, {
  FacilityFilterProps,
} from "./facilityfilter.component";
import Adapter from "enzyme-adapter-preact-pure";
import holidaysMock from "../../__mock__/holiday.json";
import { BookingResponse, Holiday } from "../../types/booking";

configure({ adapter: new Adapter() });

describe("FacilityFilter component", () => {
  const holidaysMockObj: BookingResponse = JSON.parse(
    JSON.stringify(holidaysMock)
  );
  const props: FacilityFilterProps = {
    hotelFacilities: holidaysMockObj.holidays[0].hotel.content.hotelFacilities,
    selectedFacilities: [],
    handleFacilitiesFilter: (event: Event): void => {},
    className: "",
    label: "Refine by hotel facilities",
    id: "facility-filter",
  };
  const facility_filter_component = mount(
    <FacilityFilterComponent {...props} />
  );
  it("renders properly", async () => {
    expect(facility_filter_component).toMatchSnapshot();
  });

  it("renders correct count of facilities", async () => {
    expect(
      facility_filter_component.find("#facility-filter>li").length
    ).toEqual(props.hotelFacilities.length);
  });
});
