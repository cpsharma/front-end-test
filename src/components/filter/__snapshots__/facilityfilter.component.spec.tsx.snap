// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacilityFilter component renders properly 1`] = `ReactWrapper {}`;
